blockchain/chain.go

```diff
@@ -2038,53 +2126,6 @@ func stxosToScriptSource(block *dcrutil.Block, stxos []spentTxOut, isTreasuryEna
 	return source
 }
 
-// chainQueryerAdapter provides an adapter from a BlockChain instance to the
-// indexers.ChainQueryer interface.
-type chainQueryerAdapter struct {
-	*BlockChain
-}
-
-// BestHeight returns the height of the current best block.  It is equivalent to
-// the Height field of the BestSnapshot method, however, it is needed to satisfy
-// the indexers.ChainQueryer interface.
-//
-// It is defined via a separate internal struct to avoid polluting the public
-// API of the BlockChain type itself.
-func (q *chainQueryerAdapter) BestHeight() int64 {
-	return q.BestSnapshot().Height
-}
-
-// IsTreasuryEnabled returns true if the treasury agenda is enabled as of the
-// provided block.
-func (q *chainQueryerAdapter) IsTreasuryEnabled(hash *chainhash.Hash) (bool, error) {
-	return q.IsTreasuryAgendaActive(hash)
-}
-
-// PrevScripts returns a source of previous transaction scripts and their
-// associated versions spent by the given block by using the spend journal.
-//
-// It is defined via a separate internal struct to avoid polluting the public
-// API of the BlockChain type itself.
```

Comment:
I didn't think it was worth creating a new adapter considering `Blockchain` implements a majority of the functions required by the `ChainQueryer` interface. And I wouldn't say the added functions `Best` and `PrevScripts` are polluting the public API of `Blockchain`, disregarding the `ChainQueryer` requirement they do belong in the public API of `Blockchain` I think.

Reply:
We can let others weigh-in, I mostly called it out due to the existing comments in the code.  One thing to note is that `PrevScripts` returns an indexing specific interface (`indexers.PrevScripter`), and if this is part of the public BlockChain API, it may be used for something unrelated to indexing.

Reply:
I commented on this above before seeing it was removed, but I agree with @rstaudt2.  I really don't want to see the public API polluted with things that are not specific to the blockchain.

Reply:
In fact, I ultimately would like to `blockchain` made internal, but there are some things that external callers still rely on which would to be make available via some other means.